lib/improver/tests/nowcasting/convection/handle_precip/test_ApplyPrecip.py

```diff
@@ -0,0 +1,176 @@
+
+
+class Test__repr__(IrisTest):
+
+    """Test the repr method."""
+
+    def test_basic(self):
+        """Test that the __repr__ returns the expected string."""
+        result = str(Plugin())
+        msg = ("""<ApplyPrecip:
+ precipitation mapping:
+   upper:  precip probability {precu} => max lightning prob {lprecu}
+   middle: precip probability {precm} => max lightning prob {lprecm}
+   lower:  precip probability {precl} => max lightning prob {lprecl}
+
+   heavy:  prob(precip>7mm/hr)  {pphvy} => min lightning prob {lprobl}
+   intense:prob(precip>35mm/hr) {ppint} => min lightning prob {lprobu}
+>""".format(lprobu=1., lprobl=0.25,
+            precu=0.1, precm=0.05, precl=0.0,
+            lprecu=1., lprecm=0.2, lprecl=0.0067,
+            pphvy=0.4, ppint=0.2))
+        self.assertEqual(result, msg)
+
+
+class Test_process(IrisTest):
+
+    """Test the nowcast convection handle_vii ApplyIce plugin."""
+
+    def setUp(self):
+        """Create a cube with a single non-zero point."""
+        self.fg_cube = add_forecast_reference_time_and_forecast_period(
+            set_up_cube_with_no_realizations(zero_point_indices=[]))
+        self.fg_cube.rename("probability_of_lightning")
+        self.fg_cube.coord('forecast_period').points = [0.]
+        self.precip_cube = (
+            add_forecast_reference_time_and_forecast_period(
+                set_up_cube(num_realization_points=3), fp_point=0.0))
+        threshold_coord = self.precip_cube.coord('realization')
+        threshold_coord.points = [0.5, 7.0, 35.0]
+        threshold_coord.rename('threshold')
+        threshold_coord.units = cf_units.Unit('kg m^-2')
+        self.precip_cube.rename("probability_of_precipitation")
+        self.precip_cube.attributes.update({'relative_to_threshold': 'above'})
+        self.precip_cube.data[1:, 0, ...] = 0.
+        self.precip_cube.coord('forecast_period').points = [4.]
+
+    def test_basic(self):
+        """Test that the method returns the expected cube type"""
+        plugin = Plugin()
+        result = plugin.process(self.fg_cube, self.precip_cube)
+        self.assertIsInstance(result, Cube)
+
+    def test_input(self):
+        """Test that the method does not modify the input cubes."""
+        plugin = Plugin()
+        cube_a = self.fg_cube.copy()
+        cube_b = self.precip_cube.copy()
+        plugin.process(cube_a, cube_b)
+        self.assertArrayAlmostEqual(cube_a.data, self.fg_cube.data)
+        self.assertArrayAlmostEqual(cube_b.data, self.precip_cube.data)
+
+    def test_precip_zero(self):
+        """Test that zero precip probs reduce lightning risk"""
+        plugin = Plugin()
+        expected = set_up_cube_with_no_realizations()
+        expected.data[0, 7, 7] = 0.0067
+        result = plugin.process(self.fg_cube, self.precip_cube)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+    def test_precip_small(self):
+        """Test that small precip probs reduce lightning risk"""
```

Comment:
To understand these tests we really need to know what the dimensions on the data are.  Presumably two spatial but what else?